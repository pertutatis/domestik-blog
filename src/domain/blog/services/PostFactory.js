import Post from '../model/post'
class PostFactory {
  getPost (apiData) {
    const postItem = {
      userId: apiData.userId,
      postId: apiData.id,
      date: apiData.date,
      title: apiData.title,
      lead: apiData.lead,
      body: apiData.body,
      featuredImage: apiData.featuredImage,
      images: apiData.images
    }

    return new Post(postItem)
  }
}
export default new PostFactory()
