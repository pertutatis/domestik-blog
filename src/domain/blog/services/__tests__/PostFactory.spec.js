import PostFactory from '../PostFactory'
import Post from '../../model/post'
import mockedPostFactory from './__mocks__/PostFactory'

describe('List Factory', () => {
  it('returns a List instance', () => {
    const PostInstance = PostFactory.getPost(mockedPostFactory)

    expect(PostInstance).toBeInstanceOf(Post)
    expect(PostInstance).toMatchSnapshot()
  })
})
