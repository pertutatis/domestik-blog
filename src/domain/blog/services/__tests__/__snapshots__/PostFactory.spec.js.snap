// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Factory returns a List instance 1`] = `
Post {
  "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
  "date": "01-02-2021",
  "featuredImage": "https://picsum.photos/900/506?image=1081",
  "images": Array [
    Object {
      "img": "https://picsum.photos/900/506?image=1082",
    },
    Object {
      "img": "https://picsum.photos/900/506?image=1083",
    },
  ],
  "lead": "Lorem ipsum",
  "postId": 2,
  "title": "qui est esse",
  "userId": 1,
}
`;
